admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/config/CommonConfig.java

```diff
@@ -77,5 +81,16 @@ public DefaultSort defaultSort() {
 	public AuditUtil auditUtil() {
 		return new AuditUtil();
 	}
+
+	@Bean(destroyMethod = "shutdown")
+	public ExecutorService enrichmentExecutor() {
+		final ExecutorService executorService = Executors.newFixedThreadPool(
+				Math.min(
+						Math.max(2, Runtime.getRuntime().availableProcessors()),
+						8
```

Comment:
Min(2) and Max(8) value can be taken from properties and set the default values to 2 and 8